odoo/addons/base/models/res_users.py

```diff
@@ -1685,6 +1685,14 @@ def default_get(self, fields):
         self._add_reified_groups(group_fields, values)
         return values
 
+    def _determine_fields_to_fetch(self, field_names, ignore_when_in_cache=False):
+        valid_fields = partition(is_reified_group, field_names)[1]
```

Comment:
FYI, We can use itertools.filterfalse instead of `partition` + `[1]`. 

(it's a detail, change it if you want)